From 63346cd759e08e6821622499f107ae3181f4024d Mon Sep 17 00:00:00 2001
From: Andy Yan <andy.yan@rock-chips.com>
Date: Wed, 2 Dec 2020 10:38:00 +0800
Subject: [PATCH] drm/rockchip: vop2: Change dst alpha blend mode to per pixel
 mode

This is changed by rihui.bao@rock-chips.com

Change-Id: I7d9ac1ebef93fbeff338baf0d92daf838223ac5f
Signed-off-by: Andy Yan <andy.yan@rock-chips.com>
---
 drivers/gpu/drm/rockchip/rockchip_drm_vop2.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c b/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c
index 3a0f62ae6c38..b4776798a541 100644
--- a/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c
+++ b/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c
@@ -3082,7 +3082,7 @@ static void vop2_parse_alpha(struct vop2_alpha *alpha, int pixel_alpha_en,
 	alpha->src_alpha_ctrl.bits.factor_mode = ALPHA_ONE;
 
 	alpha->dst_alpha_ctrl.bits.alpha_mode = ALPHA_STRAIGHT;
-	alpha->dst_alpha_ctrl.bits.blend_mode = alpha->dst_color_ctrl.bits.blend_mode;
+	alpha->dst_alpha_ctrl.bits.blend_mode = ALPHA_PER_PIX;
 	alpha->dst_alpha_ctrl.bits.alpha_cal_mode = ALPHA_SATURATION;
 	alpha->dst_alpha_ctrl.bits.factor_mode = ALPHA_SRC_INVERSE;
 }
-- 
2.35.3

